core/src/main/java/com/graphhopper/util/Instruction.java

```diff
@@ -40,24 +39,30 @@
     public static final int PT_START_TRIP = 101;
     public static final int PT_TRANSFER = 102;
     public static final int PT_END_TRIP = 103;
-    private static final AngleCalc AC = Helper.ANGLE_CALC;
-    protected PointList points;
     protected final InstructionAnnotation annotation;
     protected boolean rawName;
     protected int sign;
     protected String name;
     protected double distance;
     protected long time;
 
+    private int first;
+    private int last;
+
     /**
      * The points, distances and times have exactly the same count. The last point of this
      * instruction is not duplicated here and should be in the next one.
      */
-    public Instruction(int sign, String name, InstructionAnnotation ia, PointList pl) {
+    public Instruction(int sign, String name, InstructionAnnotation ia, int first) {
         this.sign = sign;
         this.name = name;
-        this.points = pl;
         this.annotation = ia;
+        this.first = first;
```

Comment:
Can we make first final?

Reply:
No, as we change the first reference during the simplification. I would have liked that as well.